invoke/config.py

```diff
@@ -808,7 +819,9 @@ def load_collection(self, data, merge=True):
         if merge:
             self.merge()
 
-    def set_project_location(self, path):
+    def set_project_location(
+        self, path: Optional[Union[PathLike, str]]
```

Comment:
Well, this specific example only uses the variable once, in os.path.join(). That function accepts `str | PathLike[str]`, so mypy is correct here as long as you add the `[str]`.

Reply:
Minor nitpick, but I'd also drop the Optional when you are already using a union and just do:
```suggestion
        self, path: Union[PathLike[str], str, None]
```

Reply:
I personally prefer to use only use one way to repres `None` but I know that's not everyone's preference. Combined all `Optinonal` / `Union`.

Reply:
Well, it's just an alias to `Union[..., None]`, so if you already have a union, it keeps it simpler, rather than nesting expressions.

Regardless, it doesn't matter long term, this will all get changed to `PathLike[str] | str | None` at a later date.

Reply:
Let me clarify: "There should be one-- and preferably only one --obvious way to do it."

What I mean by "opinion" is that `Union` looks correct for some people but `Optional` qualifies best IMO.